src/conn/conn_dhandle.c

```diff
@@ -64,16 +70,28 @@ __conn_dhandle_config_set(WT_SESSION_IMPL *session)
     switch (dhandle->type) {
     case WT_DHANDLE_TYPE_BTREE:
         WT_ERR(__wt_strdup(session, WT_CONFIG_BASE(session, file_meta), &dhandle->cfg[0]));
+        WT_ASSERT(session, dhandle->meta_base == NULL);
+        /* Overwrite any checkpoint info. */
+        cfg[1] = "checkpoint=()";
+        WT_ERR(__wt_config_collapse(session, cfg, &tmp));
+        /* Now strip out the checkpoint. */
+        cfg[0] = tmp;
+        cfg[1] = NULL;
+        WT_ERR(__wt_config_merge(session, cfg, "checkpoint=,checkpoint_lsn=", &base));
+        __wt_free(session, tmp);
         break;
     case WT_DHANDLE_TYPE_TABLE:
         WT_ERR(__wt_strdup(session, WT_CONFIG_BASE(session, table_meta), &dhandle->cfg[0]));
         break;
     }
     dhandle->cfg[1] = metaconf;
+    dhandle->meta_base = base;
```

Comment:
Have we confirmed that (except for periodic checkpoints), the file metadata never changes?

Reply:
The only way I know of is via alter. Do you know of any other paths? Or do you know of any other single path where I could confirm that?

Reply:
I agree, the only way I know is via `alter`.